packages/material-ui-icons/package.json

```diff
@@ -38,7 +38,7 @@
     "typescript": "tslint -p tsconfig.json \"src/**/*.{ts,tsx}\""
   },
   "peerDependencies": {
-    "@material-ui/core": "^4.1.2",
+    "@material-ui/core": "^4.2.0",
```

Comment:
Can we losen the version down to v4? What about using the minimum version required for the packages to work? This will prevent duplications. Updating @material-ui/styles dependencies 
 while having it the update not required has created friction in the past (duplication of the singleton).

Reply:
Since it's in `peerDependencies` it shouldn't cause any issues

Reply:
Wouldn't it be the same with a dependency? For instance a few lines above, we have:
```diff
  "dependencies": {
    "@babel/runtime": "^7.2.0",
-   "@material-ui/utils": "^4.0.1",
+   "@material-ui/utils": "^4.1.0",
```

We don't need to increase the minimal version of the utils. It works with 4.0.1. docs doesn't need any of the new features. Unless you have further insight on why the version should be increased. I think that we would be better off using the oldest possible versions. However, this is the theory, in practice, it's a different story.

When releasing new versions of the library, it's our responsibility to increase the minimal versions as they are needed by the different packages. So far, it's done manually, this opens the door for human mistakes. I haven't done any significant errors so far, but it's only a matter of time before Murphy hits us. I'm eager to put automatic checks and constraints in place to reduce the entropy of the possible cases it can go wrong. I have limited experience with Lerna, I only use it on [react-swipeable-views](https://github.com/oliviertassinari/react-swipeable-views). From what I have observed, Lerna automatically detects packages that need a version bump, and cross-reference the bump in all the packages that depend on it. So, If I understand your proposal correctly, the changes enforce the same behavior. Lerna is not the only option, I remember that necolas has a simple release script: https://github.com/necolas/react-native-web/blob/master/scripts/release/publish.js. 

So overall, it seems that we have to balance using the oldest possible version (which is critical for material-ui/styles) with reducing the scope of mistake when making releases.

Reply:
> So, If I understand your proposal correctly, the changes enforce the same behavior

The idea was more of: if core depends on `@babel/runtime@^7.2.0` then the rest of the packages should match that version so that we don't get mismatches/duplicates.

But that's not really how it works so i'll revert this change and just keep the dedupe behaviour for now